arangod/Transaction/Methods.cpp

```diff
@@ -2673,12 +2700,20 @@ Future<OperationResult> transaction::Methods::removeCoordinator(
 Future<OperationResult> transaction::Methods::removeLocal(
     std::string const& collectionName, VPackSlice value,
     OperationOptions options) {
-  auto res =
-      co_await RemoveProcessor::create(*this, collectionName, value, options);
-  if (res.fail()) {
-    co_return OperationResult(std::move(res).result(), options);
+  auto fut = Future<OperationResult>::makeEmpty();
+  // Destroy the RemoveProcessor before awaiting the future, so the
+  // BuilderLeasers return their builders before the Context might be
+  // gone.
+  {
+    auto removeProcessorRes =
+        co_await RemoveProcessor::create(*this, collectionName, value, options);
+    if (removeProcessorRes.fail()) {
+      co_return OperationResult(std::move(removeProcessorRes).result(),
+                                options);
```

Comment:
nit: options could be moved here:
```suggestion
      co_return OperationResult(std::move(removeProcessorRes).result(),
                                std::move(options));
```